docs/guides/file-system-based-automated-bundle-generation.md

```diff
@@ -0,0 +1,133 @@
+  │   └── application.js   # references ../src/my_component.js
+  │   └── application.css
+  └── src:                 # any directory name is fine. Referenced files need to be under source_path
+  │   └── component.js
+  └── stylesheets:
+  │   └── my_styles.css
+  └── images:
+      └── logo.svg
+```
+
+Now, to automatically register, `A_ComponentOne`, `B_ComponentOne`, `B_ComponentTwo` for the usage with [`react_component`](https://www.shakacode.com/react-on-rails/docs/api/view-helpers-api/#react_component) and [`react_component_hash`](https://www.shakacode.com/react-on-rails/docs/api/view-helpers-api/#react_component_hash) helpers, create directory structure as mentioned below:
+
+```
+app/javascript:
+  └── packs:                   
+  │   └── application.js     
+  │   └── application.css
+  └── src:                   
+  │   └── A
+  │   │ └── ...
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   └── A_ComponentOne.jsx
+  │   └── B
+  │   │ └── ror_components          # configured as `components_directory`
+  │   │   │ └── B_ComponentOne.jsx
+  │   │   │ └── B_ComponentTwo.jsx       
+```
+
+To register react component, it is no longer needed to create a pack entry and manually register it by calling `ReactOnRails.register`. With automatically generated packs, you can directly use `A_ComponentOne`, `B_ComponentOne` and `B_ComponentTwo` in Rails view using:
+
+```erb
+    <%= react_component("A_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentOne", {}, auto_load_bundle: true) %>    
+    <%= react_component("B_ComponentTwo", {}, auto_load_bundle: true) %>    
+```
+
+If `A_ComponentOne` uses multiple HTML strings for server rendering, [`react_component_hash`](https://www.shakacode.com/react-on-rails/docs/api/view-helpers-api/#react_component_hash) view helper can be used on Rails view as illustrated below.
+
+```erb
+<% A_ComponentOne = react_component_hash("A_ComponentOne", 
+                                             prerender: true,
+                                             auto_load_bundle: true
+                                             props: {}
+                                          ) %>   
+<% content_for :title do %>
+   <%= A_ComponentOne['title'] %>
+<% end %>
+<%= A_ComponentOne["componentHtml"] %>
+```
+
+the default value of the auto_load_bundle parameter can be specified by setting `config.auto_load_bundle` in `config/initializers/react_on_rails.rb`.
+
+### Server Rendering and Client Rendering Components
+
+If server rendering is enabled, the component will be registered for the usage both on server and client rendering. In order to have separate definitions for client and server rendering, name the components files as `Component_Name.server.jsx` and `Component_Name.client.jsx`. During the server rendering, the `Component_Name.server.jsx` definition will be used, and for client rendering, `Component_Name.client.jsx` file will be used.
```

Comment:
2. "for the usage both on" I think "the" should be removed. And less sure, but "in" instead of "on"? Without "rendering" it would be "on", yes.
3. "the components files" -> "the component files"?
4. "During the server rendering, the `Component_Name.server.jsx` definition will be used, and for client rendering, `Component_Name.client.jsx` file will be used" Why "the" for server, but not for client? I think 

    > The `Component_Name.server.jsx` file will be used for server rendering, and the `Component_Name.client.jsx` file for client rendering.

    Sounds good?
1. From the test I can see that only having `.client.jsx` is allowed (when you want not to render on server at all, right?). If that's correct, let's document it here. E.g. 

     > If you don't want the component to be rendered on server, you should only have the `Component_Name.client.jsx` file.